ksqldb-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -165,6 +166,7 @@ public long read() {
 
   public void initialize() {
     // initialize the first KafkaStreams
+    MDC.put("queryId", queryId.toString());
```

Comment:
I think the MDC is thread-local - so we'd need this to be done from the streams thread itself.

Reply:
https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/MDC.html

It is but `The MDC is managed on a per thread basis. A child thread automatically inherits a copy of the mapped diagnostic context of its parent.`